src/renderer/containers/QuickStart/VerifyEmailStep.tsx

```diff
@@ -0,0 +1,138 @@
+
+import { QuickStartHeader } from "./QuickStartHeader";
+
+const Container = styled.div`
+  margin: 0 auto;
+  width: 100%;
+  max-width: 800px;
+`;
+
+const classes = {
+  message: css`
+    color: ${colors.textSecondary};
+  `,
+  emailContainer: css`
+    background: ${colors.purpleDark};
+    padding: 10px 20px;
+    border-radius: 5px;
+    margin: 10px;
+    width: fit-content;
+    font-size: 20px;
+  `,
+  incorrectEmailContainer: css`
+    margin-top: -5px;
+    margin-left: 10px;
+    font-size: 14px;
+    color: ${colors.textDim};
+    > a {
+      color: ${colors.purplePrimary};
+    }
+  `,
+  instructions: css`
+    margin-top: 25px;
+    margin-bottom: 10px;
+    color: ${colors.textSecondary};
+  `,
+  emailNotFoundContainer: css`
+    margin-top: 4px;
+    font-size: 14px;
+    color: ${colors.textDim};
+    > a {
+      color: ${colors.purplePrimary};
+    }
+  `,
+  confirmationContainer: css`
+    margin-top: 30px;
+    font-size: 28px;
+    color: ${colors.greenPrimary};
+    display: grid;
+    align-items: center;
+    gap: 8px;
+    grid-template-columns: auto auto 1fr;
+    > svg {
+      font-size: 40px;
+    }
+  `,
+};
+
+export const VerifyEmailStep: React.FC = () => {
+  const { authService } = useServices();
+  const setServerError = useAccount((store) => store.setServerError);
+  const setUser = useAccount((store) => store.setUser);
+  const user = useAccount((store) => store.user);
+  const emailVerificationSent = useAccount((store) => store.emailVerificationSent);
+  const setEmailVerificationSent = useAccount((store) => store.setEmailVerificationSent);
+
+  const profileUrl = "https://slippi.gg/profile";
+
+  const handleCheckVerification = async () => {
+    try {
+      const newUser = await authService.reloadUser();
+      setUser(newUser);
+    } catch (err: any) {
+      setServerError(err.message);
+    }
+  };
+
+  useEffect(() => {
+    void (async () => {
+      if (user && !user.emailVerified && !emailVerificationSent) {
+        try {
+          await authService.verifyEmail();
+          setEmailVerificationSent(true);
+        } catch (err: any) {
+          setServerError(err.message);
+        }
+      }
+    })();
+  }, [emailVerificationSent, setEmailVerificationSent, setServerError, user, authService]);
+
+  const preVerification = (
+    <>
+      <div css={classes.instructions}>
+        Visit your email, click the link in the verification email, and then check verification
+      </div>
+      <Button variant="outlined" onClick={handleCheckVerification}>
+        Check Verification
+      </Button>
+      <div css={classes.emailNotFoundContainer}>
+        Not finding email?{" "}
+        <a href="#" onClick={authService.verifyEmail}>
+          send again
+        </a>
+      </div>
+    </>
+  );
+
+  const postVerification = (
+    <div css={classes.confirmationContainer}>
+      <CheckCircleOutlineIcon />
+      Email verified
+    </div>
+  );
+
```

Comment:
```suggestion
  if (!user) {
    return null; // Or a message of some sort
  }
```
Can we handle the null user case earlier to avoid having to do optional chaining below?